crates/external_services/src/file_storage.rs

```diff
@@ -0,0 +1,124 @@
+//!
+//! Module for managing file storage operations with support for multiple storage schemes.
+//!
+
+use std::fmt::{Display, Formatter};
+
+use common_utils::{
+    errors::{CustomResult, FileStorageError},
+    fs_utils,
+};
+use error_stack::ResultExt;
+
+#[cfg(feature = "aws_s3")]
+use crate::file_storage::aws_s3::{AwsFileStorageClient, AwsFileStorageConfig};
+
+/// When the "aws_s3" feature is enabled, the module includes functionality for AWS S3 storage.
+#[cfg(feature = "aws_s3")]
+pub mod aws_s3;
+
+/// Enum representing different file storage configurations, allowing for multiple storage schemes.
+#[derive(Debug, Clone, Default, serde::Deserialize)]
+#[serde(tag = "file_storage_backend")]
+#[serde(rename_all = "snake_case")]
+pub enum FileStorageConfig {
+    /// AWS S3 storage configuration.
+    #[cfg(feature = "aws_s3")]
+    AwsS3 {
+        /// Configuration for AWS S3 file storage.
+        aws_s3: AwsFileStorageConfig,
+    },
+    /// Local file system storage configuration.
+    #[default]
+    FileSystem,
+}
+
+impl FileStorageConfig {
+    /// Validates the file storage configuration.
+    pub fn validate(&self) -> Result<(), InvalidFileStorageConfig> {
+        match self {
+            #[cfg(feature = "aws_s3")]
+            Self::AwsS3 { aws_s3 } => aws_s3.validate(),
+            Self::FileSystem => Ok(()),
+        }
+    }
+}
+
+/// Enum representing different file storage clients.
+#[derive(Debug, Clone)]
+pub enum FileStorageScheme {
```

Comment:
We might have to call this `FileStorageBackend`, to go with the name of the configuration variable.